add a failing test for and fix deposit_address class being wrong

andrewroth opened the pull request:
it was RestClient::Response, it should be String (github issue #25)

andrewroth committed:
add a failing test for and fix deposit_address class being wrong

it was RestClient::Response, it should be String (github issue #25)

kevinjalbert commented on lib/quadrigacx/request.rb:
```diff
@@ -26,7 +26,7 @@ def fix_json response
       return true  if response.strip == '"true"'
       return false if response.strip == '"false"'
-      response[/"(.*)"/, 1] || response
+      response.to_s[/"(.*)"/, 1] || response
```
So this is an interesting method -- `fix_json`. As the comment says, it was original for the `/cancel_order` API endpoint which doesn't return valid JSON.

We're seeing the same thing now with the deposit address -- just getting back a string of an address.

- [ ] I would first update the comment to also account for this new situation, or generalize it to address _responses that cannot be parsed as JSON_. 

While the `response` is a `RestClient::Response`, it still inherits from `String` (http://www.rubydoc.info/gems/rest-client/RestClient/Response). This means that it technically still worked and behaved as a string. I do see your concern if you are using this gem externally -- you would expect it to be straight-up a `String`.

- [ ] I suggest we change the returned output to a `String`, but instead, take the preferred path of using `response.body`:

![](https://monosnap.com/file/ZdNyj5cOytrKZ8aRhm2eLr3f8ieP2I.png)

- [ ] To be future proofed we should be using `response.body` even before the `.strip`. I would just make a new variable `response_body` (something like that) and use that throughout the rest of this method.

kevinjalbert replied:
Good catch on this by the way!

andrewroth committed:
fix_json improvements

explain fix_json in generalized terms instead of just for cancel order

use response.body as the preferred way to get a string from
RestClient::Response

andrewroth commented:
Good suggestions, how does this look?

kevinjalbert commented on lib/quadrigacx/request.rb:
```diff
@@ -23,10 +23,11 @@ def request *args
-      return true  if response.strip == '"true"'
-      return false if response.strip == '"false"'
-      response.to_s[/"(.*)"/, 1] || response
+      # This handles all responses that cannot be parsed as JSON
```
Maybe just one more change here ` -- generally of a form "xxxxxx"`. Just because that is what we're expecting within this method.

andrewroth committed:
expand on fix_json comment a bit

andrewroth commented:
Ok, how's this?

kevinjalbert reviewed (approved):
👌  from my end.
